src/video.cpp

```diff
@@ -539,8 +539,8 @@ std::pair<float, float> CVideo::get_dpi_scale_factor() const
 	if(dpi.first != 0.0f && dpi.second != 0.0f) {
 		// adjust for pixel scale
 		SDL_Point wsize = window_size();
-		dpi.first /= wsize.x / get_width();
-		dpi.second /= wsize.y / get_height();
+		dpi.first *= float(get_width()) / wsize.x;
```

Comment:
Only thing is `static_cast` is preferred to C-style casts but that can be cleaned up separately. If this PR fixes the issue I say merge ASAP.

Reply:
This is technically "constructor-style", not "C-style". Not that that changes your point substantially.

Reply:
Is there something that could be set to catch these by the compiler? For C style casts for example, there's `-Wold-style-cast`.

Reply:
`static_cast` for a basic numeric type? Why? That just makes code messier and more difficult to read.

Reply:
I just checked and `-Wold-style-cast` is already set and does indeed warn if you try to do something dodgy.

Reply:
Well, personally I prefer constructor-style casts for this type of thing. Clearly Vultraz (and some other people) disagree.

Reply:
FTR I don't really have an opinion either way, I just prefer enforcing things as much as possible in the CI so that it's kept consistent - assuming this is meant to be a coding convention of Wesnoth's code base.

Reply:
I don't think this makes sense as a coding convention. If the CI treats warnings as errors it will already catch dangerous casts, such as casting pointers or structs.

In cases like the one here it is an extra 13 characters (including angle brackets) each use for no benefit.